### PR TITLE
Adding test for "GET /images/(name)/history" API.

### LINKED ISSUE #12284: Add History API test
https://github.com/docker/docker/pull/12260/files deleted a test, this needs to be reimplemented in integration-cli

### PR DESCRIPTION
**Pull request by draghuram**
Closes #12284.

Signed-off-by: Raghuram Devarakonda draghuram@gmail.com

### CODE REVIEW

File: integration-cli/docker_api_images_test.go [hunk 1]
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
...
+		c.Fatal(err)
+	}
+	defer deleteImages(name)
+	id := strings.TrimSpace(out)
+
+	status, body, err := sockRequest("GET", "/images/"+id+"/history", nil)
+	c.Assert(status, check.Equals, http.StatusOK)
+	c.Assert(err, check.IsNil)
+
+	var historydata []map[string]interface{}
```
Comment (runcom) [inline integration-cli/docker_api_images_test.go]: Please decode in this struct https://github.com/docker/docker/blob/master/api/types/types.go#L51

File: integration-cli/docker_api_images_test.go [hunk 2]
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
 	c.Assert(status, check.Equals, http.StatusOK)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestApiImagesHistory(c *check.C) {
```
Comment (runcom) [inline integration-cli/docker_api_images_test.go]: you need to add `testRequires(c, Network)` in order for this test to work on --net none build

File: integration-cli/docker_api_images_test.go [hunk 3]
```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
...
+	name := "test-api-images-history"
+	out, err := buildImage(name, "FROM hello-world\nENV FOO bar", false)
+	if err != nil {
+		c.Fatal(err)
+	}
```
Comment (runcom) [inline integration-cli/docker_api_images_test.go]: nit, you can use `c.Assert(err, check.IsNil`)

**Comment by draghuram** [comment]
@runcom Thanks for the review. This is my first foray into go world so any suggestions to improve code are welcome. 

I have made following changes:

- Added `testRequires(c, Network)`
- Replaced explicit error checking with Asserts.
- Added an additional assertion for the returned data (in addition to its size).

**Comment by runcom** [comment]
@draghuram  thanks!

**Comment by tiborvass** [comment]
LGTM

**Comment by estesp** [comment]
LGTM

**Pull request merged**